### PR TITLE
nudb reporting tool

### PR DESCRIPTION
**Pull request by vinniefalco**
This reports statistics on rippled-specific contents of a nudb dat file.

### CODE REVIEW

File: src/ripple/nodestore/tests/dump_test.cpp [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+
+    using prefix_t = std::array<char, 3>;
+
+    static
+    std::string
+    to_string(bucket const& b, std::string const& name)
+    {
+        std::string s;
+        s = std::to_string(b.count) + " " + name + " in " + std::to_string(b.bytes) + " bytes";
+        return s;
```
Comment (HowardHinnant) [inline src/ripple/nodestore/tests/dump_test.cpp]: Why not just:
````
return std::to_string(b.count) + " " + name + " in " + std::to_string(b.bytes) + " bytes";
````

File: src/ripple/nodestore/tests/dump_test.cpp [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+                "path:   NuDB path to database (without the .dat)\n";
+                "every:  Intermediate report every # items (0 to disable)\n";
+            return;
+        }
+
+        auto const path = args.at("path");
+
+        std::size_t every = 1000000;
+        if(args.find("every") != args.end())
+            every = std::stoull(args.at("every"));
```
Comment (HowardHinnant) [inline src/ripple/nodestore/tests/dump_test.cpp]: Or:
````
        std::size_t const every = args.find("every") != args.end() ?
                                      std::stoull(args.at("every")) :
                                      1'000'000;
````

**Comment by nbougalis** [comment]
Looks reasonable :+1:

**Comment by HowardHinnant** [comment]
👍  Looks good to me.  A couple nitpick (optional) comments.

File: src/ripple/nodestore/tests/dump_test.cpp [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+                "Missing parameter: path";
+            usage = true;
+        }
+
+        if (usage)
+        {
+            log <<
+                "Usage:\n" <<
+                "--unittest-arg=path=<path>[,every=<number>]\n" <<
+                "path:   NuDB path to database (without the .dat)\n";
```
Comment (HowardHinnant) [inline src/ripple/nodestore/tests/dump_test.cpp]: Mistaken `;` at end instead of ` <<`.

File: src/ripple/nodestore/tests/util.h [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+#include <ripple/basics/contract.h>
+#include <beast/http/rfc2616.hpp>
+#include <beast/detail/ci_char_traits.hpp>
+#include <boost/regex.hpp>
+#include <map>
+#include <string>
+
+namespace ripple {
+namespace NodeStore {
+
```
Comment (HowardHinnant) [inline src/ripple/nodestore/tests/util.h]: `parse_args` either needs to be moved to a source, or needs to be given weak linkage (inline or gratuitous template).  I'm getting duplicate symbol linker error without such treatment.

File: src/ripple/nodestore/tests/dump_test.cpp (hunk 2 above)
Comment (vinniefalco) [inline src/ripple/nodestore/tests/dump_test.cpp]: noice!

File: src/ripple/nodestore/tests/util.h (hunk 4 above)
Comment (vinniefalco) [inline src/ripple/nodestore/tests/util.h]: Too much writing header-only code and I've forgotten how to write normal code

**Comment by HowardHinnant** [comment]
👍  lgtm.

**Comment by vinniefalco** [comment]
No need to merge this yet.

**Comment by nbougalis** [comment]
What are we doing with this @vinniefalco?

**Comment by vinniefalco** [comment]
Nothing I guess

**Pull request closed without merging**